Source/DependencyInversion/TenantScopedProviders.cs

```diff
@@ -13,15 +14,15 @@ namespace Dolittle.SDK.DependencyInversion;
 /// </summary>
 public class TenantScopedProviders : ITenantScopedProviders
 {
-    readonly IReadOnlyDictionary<TenantId, IServiceProvider> _serviceProviders;
+    readonly IReadOnlyDictionary<TenantId, (AutofacServiceProvider, IDisposable)> _serviceProviders;
 
     /// <summary>
     /// Initializes a new instance of the <see cref="TenantScopedProviders"/> class.
     /// </summary>
     /// <param name="tenantScopedServiceProviders">The <see cref="IDictionary{TKey,TValue}"/> of <see cref="IServiceProvider"/> per <see cref="TenantId" />.</param>
-    public TenantScopedProviders(IDictionary<TenantId, IServiceProvider> tenantScopedServiceProviders)
+    public TenantScopedProviders(IDictionary<TenantId, (AutofacServiceProvider, IDisposable)> tenantScopedServiceProviders)
     {
-        _serviceProviders = new ReadOnlyDictionary<TenantId, IServiceProvider>(tenantScopedServiceProviders);
+        _serviceProviders = new ReadOnlyDictionary<TenantId, (AutofacServiceProvider, IDisposable)>(tenantScopedServiceProviders);
```

Comment:
I'm not super excited by the actual implementations leaking out. I would be more excited if the loop down below here just did a foreach, and maybe something like `if (provider is Disposable ...)`. I also have to admit that I really don't like the anonymous tuple `.Item1` and `.Item2` things.

Reply:
I was a bit unsure how we wanted this, nice to get some feedback. I agree about the Item# part